src/librustc_mir/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -1225,8 +1204,9 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         );
 
         let msg = match category {
-            ConstraintCategory::Return => "closure is returned here".to_string(),
-            ConstraintCategory::OpaqueType => "generator is returned here".to_string(),
+            ConstraintCategory::Return | ConstraintCategory::OpaqueType => {
```

Comment:
I feel the existing behavior is wrong, users of async should not see `generator` in error messages.
But not sure what wording should be used instead.

Reply:
I think that we'll 1) need to be able to differentiate between closures and generators here in the first place and 2) in the `OpaqueType` it should be `"async closure is returned here"`, right?

Reply:
Agreed that saying generators in an error message is basically never right, regardless.